src/views/Home.jsx

```diff
@@ -18,12 +22,51 @@ export function Home({ listToken, setListToken }) {
 		}
 	}, [listToken]);
 
+	const handleChange = (e) => {
+		setExistingToken(e.target.value);
+	};
+
+	const handleSubmit = (e) => {
+		e.preventDefault();
+
+		// conects to db
+		// checks if the token that user inputted is associated with any lists
+		streamListItems(existingToken, (snapshot) => {
+			const nextData = getItemData(snapshot);
+
+			// if there is a list, set the token to the listToken state
+			if (nextData.length > 0) {
+				setListToken(existingToken);
+			} else {
+				setErrorMessage("List doesn't exist");
+			}
+		});
+	};
 	return (
 		<div className="Home">
 			<p>
 				Hello from the home (<code>/</code>) page!
 			</p>
-			{!listToken && <button onClick={createToken}>Create a new list</button>}
+
+			<div>
+				<button onClick={createToken}>Create a new list</button>
+
+				<form onSubmit={handleSubmit}>
+					<label htmlFor="existingToken">
+						Join an existing list
+						<input
+							type="text"
+							id="existingToken"
+							value={existingToken}
+							onChange={handleChange}
+						/>
+					</label>
+
+					<button type="submit">Submit</button>
+				</form>
+
+				{errorMessage.length > 0 && <p>{errorMessage}</p>}
```

Comment:
Since an empty string is falsy, we can refactor this line to just check if `errorMessage` is a truthy value as this will ensure that it is not an empty string.
```suggestion
				{errorMessage && <p>{errorMessage}</p>}
```